### PR TITLE
Add changelog and version

### PR DESCRIPTION
**Pull request by ktp242**
This PR just updates the changes on CHANGELOG and bumps the version. It now points to `development` as the base, but it will wait until all other PRs to be merged, so it will then be merged to the latest `development`.

It also fixes some syntax issue that eslint complains.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 ## CHANGELOG
 
+### 4.0.0
+> Update the routes. The basic route is now `/books-music-movies/recommendations/best-books/`. And `staff-picks`, `ya`, `childrens` are three lists on the same level of the structure.
```
Comment (EdwinGuzman) [inline CHANGELOG.md]: `base` not `basic`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
...
+> Add the list selectors that help the user to choose the list by season/month or audience.
+> Add individual page for every single book for supporting legacy staff-picks links and homepage's staff-picks section. There won't be links to those pages from the main page.
```
Comment (EdwinGuzman) [inline CHANGELOG.md]: We converted the modal page into a static page, we didn't "add" any new feature for this.

Comment (EdwinGuzman) [reply CHANGELOG.md]: Hmm actually, I think `update` is a better word than `add` or `convert`.

File: src/server/routes/monthData.js [hunk 3]
```diff
@@ -90,8 +89,9 @@ function selectMonthData(req, res, next) {
   if (!seasonMatches || !isValidAudience) {
     console.error('Status Code: 400, Error Message: Invalid season or audience.');
 
-    return; res.redirect('/books-music-dvds/recommendations/404');
-  } else {
+    return res.redirect('/books-music-dvds/recommendations/404');
```
Comment (holingpoon) [inline src/server/routes/monthData.js]: `'/books-music-movies/recommendations/404'` here?

File: src/server/routes/monthData.js [hunk 4]
```diff
@@ -144,7 +143,6 @@ function selectMonthData(req, res, next) {
 
       return res.redirect('/books-music-dvds/recommendations/404');
```
Comment (holingpoon) [inline src/server/routes/monthData.js]: Also '/books-music-movies/recommendations/404' here?

File: src/server/routes/monthData.js [hunk 5]
```diff
@@ -200,7 +198,8 @@ function selectMonthDataFormPost(req, res) {
 
     res.redirect('/books-music-dvds/recommendations/404');
```
Comment (holingpoon) [inline src/server/routes/monthData.js]: '/books-music-movies/recommendations/404' here?

File: src/server/routes/monthData.js [hunk 6]
```diff
@@ -200,7 +198,8 @@ function selectMonthDataFormPost(req, res) {
...
   } else {
-    // Redirects and calls selectMonthData() to make server side request for the season/audience list
+    // Redirects and calls selectMonthData() to make server side request
+    // for the season/audience list
     res.redirect(
       `${config.baseMonthUrl}${season}${audienceQuery}`
```
Comment (holingpoon) [inline src/server/routes/monthData.js]: `${config.baseUrl}staff-picks/${season}${audienceQuery}` here?

File: src/server/routes/monthData.js (hunk 3 above)
Comment (ktp242) [reply src/server/routes/monthData.js]: This should be updated once your PR is merged to `development` and I merge from new `development` again.

Comment (holingpoon) [reply src/server/routes/monthData.js]: I see. :+1:

File: test/unit/DateService.test.js [hunk 7]
```diff
@@ -3,20 +3,22 @@ import { expect } from 'chai';
 
 import DateService from '../../src/app/utils/DateService';
 
-describe('When no date is passed', () => {
-  it('should return the object that has empty month and year values.', () => {
-    expect(DateService(undefined)).to.deep.equal({ month: '', year: '' });
+describe('DateService', () => {
```
Comment (ktp242) [inline test/unit/DateService.test.js]: Wrap the tests within the service's name for better readability.

File: test/unit/ModelListOptionsService.test.js [hunk 8]
```diff
@@ -24,34 +24,36 @@ const mockModeledOptionsNotStaffPicks = [
   { name: '', value: '2015-04-01' },
 ];
 
-describe('When there is no data', () => {
-  it('should return the object, { options: [], latestOption: \'\' }', () => {
-    expect(ModelListOptionsService(undefined, 'staff-picks')).to.deep.equal(
-      { options: [], latestOption: '' }
-    );
+describe('ModelListOptionsService', () => {
```
Comment (ktp242) [inline test/unit/ModelListOptionsService.test.js]: Wrap the tests within the service's name for better readability.

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,11 @@
 ## CHANGELOG
 
+### 4.0.0
+> Update the routes. The base route is now `/books-music-movies/recommendations/best-books/`. And `staff-picks`, `ya`, `childrens` are three lists on the same level of the structure.
+> Add the list selectors that help the user choose the list by season/month or audience.
+> Update an individual page for every single book for supporting legacy staff-picks links and homepage's staff-picks section. There won't be links to those pages from the main page.
```
Comment (gkallenberg) [inline CHANGELOG.md]: Maybe "Update the BookPage component to support legacy staff-picks direct links and homepage's staff-picks component as the modal replacement from the previous staff-picks application. No direct links to picks in lists is supported at this time."  ??

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,11 @@
...
+> Add an `.env.example` and update the related README.md so now the user can set a `.env` to run the application locally.
```
Comment (gkallenberg) [inline CHANGELOG.md]: Change "user" to "developer"

File: CHANGELOG.md (hunk 9 above)
Comment (ktp242) [reply CHANGELOG.md]: sure !

**Pull request merged**